### PR TITLE
Update utils_test.dart to reflect renaming of the tab appsize to app-size

### PR DESCRIPTION
**Pull request by dkbast**
This is a hygiene commit, cleaning up after https://github.com/dart-lang/sdk/commit/cc309f25d5d9cf6e402368f3b74b7fe86a8eedb5 which solves https://github.com/flutter/devtools/issues/7173 but is a leftover which was not cleaned up.

Updating the url, even tough the url does not have any effect in this test. This is mostly a cleanup as this is the last place where this is still the old path and it was confusing when I tried finding the issue described in #7173 

Tagging @kenzieschmoll for review.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x]  there is a reason for not adding tests.

![build.yaml badge]

If you need help, consider asking for help on [Discord].

[Contributor Guide]: https://github.com/flutter/devtools/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[build.yaml badge]: https://github.com/flutter/devtools/actions/workflows/build.yaml/badge.svg

### CODE REVIEW

**Review by kenzieschmoll — APPROVED** [review]

**Comment by dkbast** [comment]
I thinks this qualifies for label:release-notes-not-required

**Comment by kenzieschmoll** [comment]
Looks like there are some failing tests. Can you try to merge with master and see if that fixes the failures?

**Comment by dkbast** [comment]
@kenzieschmoll the parsing of query parameters seems to have been broken - this should fix it (regex was not matching hyphens, since '\w' only matches agains letters, digits and underscores), we probably should add a test for all existing devtools paths, to check that they can receive query parameters.

**Comment by kenzieschmoll** [comment]
The golden failures are fixed by a PR I just landed. If you merge master again, that failure will go away.

File: packages/devtools_app/lib/src/shared/primitives/utils.dart [hunk 1]
```diff
@@ -1167,7 +1167,7 @@ String simplifyDevToolsUrl(String url) {
   // http://localhost:123/#/page-id?key=value
   // Since we just want the query params, we will modify the url to have an
   // easy-to-parse form.
-  return url.replaceFirst(RegExp(r'#\/(\w*)[?]'), '?');
+  return url.replaceFirst(RegExp(r'#\/([\w\-]*)[?]'), '?');
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/primitives/utils.dart:1170]: if we are changing this method, we should have a test that verifies this still gives us the expected behavior. The `app-size` screen is not the only page with a hyphen in the name, so I'm surprised that we haven't seen broken behavior before. So would be good to test this thoroughly.

Comment (dkbast) [reply packages/devtools_app/lib/src/shared/primitives/utils.dart:1170]: I just updated the test to iterate over all existing tabs

File: packages/devtools_app/test/primitives/utils_test.dart [hunk 2]
```diff
@@ -666,15 +667,17 @@ void main() {
...
-        ),
-        equals({
-          'key': 'value.json',
-          'key2': '123',
-        }),
-      );
+      for (final meta in ScreenMetaData.values) {
+        expect(
+          devToolsQueryParams(
+            'http://localhost:9101/#/${meta.id}?key=value.json&key2=123',
```
Comment (kenzieschmoll) [inline packages/devtools_app/test/primitives/utils_test.dart:673]: nit: you can remove the `/#` fragment too now. That is not part of the DevTools URL anymore.

**Review by kenzieschmoll — APPROVED** [review]
LGTM with one nit comment. Thanks!

File: packages/devtools_app/test/primitives/utils_test.dart (hunk 2 above)
Comment (dkbast) [reply packages/devtools_app/test/primitives/utils_test.dart]: 👍

**Comment by dkbast** [comment]
Fingers crossed 🙃

**Pull request merged**